Clarify ECE - document host maintenance with podman (#3066)
elastic/docs-content: deploy-manage/maintenance/ece/perform-ece-hosts-maintenance.md

You can perform these maintenance actions on the hosts in your ECE installation using one of these methods:

* [By disabling the Docker daemon (nondestructive)](#ece-perform-host-maintenance-docker-disable)

> * [By disabling the Podman related services (nondestructive)](#ece-perform-host-maintenance-podman-disable) - **For Podman Only** 

Reviewer: ```suggestion
* [By disabling the Podman related services (nondestructive)](#ece-perform-host-maintenance-podman-disable) 
```

* [By deleting the host (destructive)](#ece-perform-host-maintenance-delete-runner)
* [By shutting down the host (less destructive)](#ece-perform-host-maintenance-delete-runner)

Which method you choose depends on how invasive your host maintenance needs to be. If your host maintenance could affect ECE, use the destructive method that first deletes the host from your installation. These methods include a step that moves any hosted {{es}} clusters and {{kib}} instances off the affected hosts and are generally considered safe, provided that your ECE installation still has sufficient resources available to operate after the host has been removed.

> ## By disabling the Docker daemon (Non Destructive) [ece-perform-host-maintenance-docker-disable]

Reviewer: always use sentence case
```suggestion
## By disabling the Docker daemon (nondestructive) [ece-perform-host-maintenance-docker-disable]
```

After the host shows a green status in the Cloud UI, it is fully functional again and can be used as before.

> ## By disabling the Podman related services (Non Destructive) [ece-perform-host-maintenance-podman-disable]

Reviewer: ```suggestion
## By disabling the Podman related services (nondestructive) [ece-perform-host-maintenance-podman-disable]
```

:::{note}
This section only applies to Podman. 

> :::

Reviewer: wouldn't it be correct that the docker steps are only for docker installations? 

suggest that the title makes tihs clear

Reviewer 2: >wouldn't it be correct that the docker steps are only for docker installations?

yep.  If making that sort of note under podman, might it make sense to do the same under docker?

What if the page is also reorganized in this fashion to group by method/intent?

>- By disabling the container services (nondestructive):
>    - For Docker-based installations: Disabling the Docker service
>    - For Podman-based installations: Disabling the Podman-related services
>- By deleting the host (destructive)
>- By shutting down the host (less destructive)

This method lets you perform maintenance actions on hosts without first removing the associated host from your {{ece}} installation. It works by disabling the Podman related services. The host remains a part of your ECE installation throughout these steps but will be offline and the resources it provides will not be available.

To perform host maintenance:

1. Recommended: If the host holds the allocator role and you have enough spare capacity:
   1. [Enable maintenance mode](enable-maintenance-mode.md) on the allocator.
   2. [Move all nodes off the allocator](move-nodes-instances-from-allocators.md) and to other allocators in your installation. Moving all nodes lets you retain the same level of redundancy for highly available {{es}} clusters and ensures that other clusters without high availability remain available.
   ::::{important}
   Skipping Step 1 will affect the availability of clusters with nodes on the allocator.
   ::::

> 2. Disable the Podman Service, Podman Socket, and Podman Restart Service

Reviewer: ```suggestion
2. Disable the Podman service, Podman socket, and Podman restart service:
```

    ```sh
    sudo systemctl disable podman.service
    sudo systemctl disable podman.socket
    sudo systemctl disable podman-restart.service
    ```

3. Reboot the host:

    ```sh
    sudo reboot
    ```

> After rebooting, confirm there are no running containers:
> - `sudo podman ps` - Output should be empty
> 
> If an `frc-*` or `fac-*` container is found to be running for some reason, stop it:
> - `sudo podman stop $(sudo podman ps -a --filter "name=fac" --filter "name=frc" --format "{{.ID}}")`

Reviewer: ````suggestion
4. After rebooting, confirm there are no running containers by running the following command. The output should be empty.
    ```sh
    sudo podman ps
    ```

	If an `frc-*` or `fac-*` container is returned in the output, stop it:
	
	```sh
	sudo podman stop $(sudo podman ps -a --filter "name=fac" --filter "name=frc" --format "{{.ID}}")
	```
````

4. Perform your maintenance on the host, such as patching the operating system.
5. Re-enable the Podman related services:

    ```sh
    sudo systemctl enable podman.service
    sudo systemctl enable podman.socket
    sudo systemctl enable podman-restart.service
    ```

6. Reboot the host again:

    ```sh
    sudo reboot
    ```

> Confirm the containers have started:
> - `sudo podman ps -a`
>   - use `-a` flag so no containers are overlooked 

Reviewer: ````suggestion
7. Confirm the containers have started:

    ```sh
    sudo podman ps -a
    ```
  
  The use `-a` flag ensures that no containers are overlooked.
````

> 7. If you enabled maintenance mode in Step 1: Take the allocator out of maintenance mode.

Reviewer: ```suggestion
7. If you enabled maintenance mode in Step 1, take the allocator out of maintenance mode.
```
